outlines/fsm/fsm.py

```diff
@@ -182,6 +182,47 @@ def next_state(self, state: FSMState, token_id: int) -> FSMState:
 
         return FSMState(next_state)
 
+    @classmethod
+    def from_interegular_fsm(
+        cls, interegular_fsm: interegular.fsm.FSM, tokenizer: "Tokenizer"
+    ):
+        from_interegular_instance = cls.__new__(cls)
+
+        @cache()
+        def create_states_mapping_from_interegular_fsm(
```

Comment:
As discussed here, `@cache` doesn't work with FSMs. 

https://github.com/outlines-dev/outlines/issues/670#issuecomment-1958266151

We should leave this uncached for now. There needs to be an upstream fix in `interegular` which makes `AnythingElse` have a consistent hash before we can cache.

We should leave this uncached for now.